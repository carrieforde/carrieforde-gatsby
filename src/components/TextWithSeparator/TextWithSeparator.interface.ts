export interface TextWithSeparatorProps {
  text1: string;
  text2: string;
  separator?: string;
}
