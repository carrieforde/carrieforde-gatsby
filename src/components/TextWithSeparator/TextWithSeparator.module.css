.textWithSeparator {
	color: var(--heading-color-alt);
	display: inline-block;
	font-size: var(--font-size-1);
	font-weight: 300;
	margin-bottom: var(--size-2);
	margin-top: var(--size-2);
}

.textWithSeparator__separator {
	margin-left: var(--size-1);
	margin-right: var(--size-1);
}
