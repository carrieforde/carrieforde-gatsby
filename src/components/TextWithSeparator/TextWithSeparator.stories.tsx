import React from 'react';
import TextWithSeparator from './TextWithSeparator';
import { TextWithSeparatorProps } from './TextWithSeparator.interface';
import { Story, Meta } from '@storybook/react';

export default {
  title: 'TextWithSeparator',
  component: TextWithSeparator,
} as Meta;

const Template: Story<TextWithSeparatorProps> = (args) => (
  <TextWithSeparator {...args} />
);

export const Default = Template.bind({});
export const WithCustomSeparator = Template.bind({});

Default.args = {
  text1: 'San Mateo, CA',
  text2: 'June 2018 &ndash; Present',
};

WithCustomSeparator.args = {
  text1: 'March 08, 2021',
  text2: '9 minute read',
  separator: '&vert;',
};
