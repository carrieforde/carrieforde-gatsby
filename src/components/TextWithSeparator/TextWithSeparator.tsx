import React from 'react';
import { TextWithSeparatorProps } from './TextWithSeparator.interface';
import parse from 'html-react-parser';
import {
  textWithSeparator,
  textWithSeparator__separator,
} from './TextWithSeparator.module.css';

const TextWithSeparator: React.FC<TextWithSeparatorProps> = ({
  text1,
  text2,
  separator = '&#9656;',
}) => {
  return (
    <p className={textWithSeparator}>
      <span>{parse(text1)}</span>
      <span className={textWithSeparator__separator}>{parse(separator)}</span>
      <span>{parse(text2)}</span>
    </p>
  );
};

export default TextWithSeparator;
