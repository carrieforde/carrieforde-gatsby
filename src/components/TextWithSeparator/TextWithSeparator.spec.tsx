import { render } from "@testing-library/react";
import React from "react";
import { Default, WithCustomSeparator } from "./TextWithSeparator.stories";

describe("TextWithSeparator", () => {
  it("should render the TextWithSeparator component", () => {
    const { container } = render(<Default {...Default.args} />);
    expect(container).toMatchSnapshot();
  });

  it("should render a TextWithSeparator with a custom separator", () => {
    const { container } = render(
      <WithCustomSeparator {...WithCustomSeparator.args} />
    );

    expect(container).toMatchSnapshot();
  });
});
