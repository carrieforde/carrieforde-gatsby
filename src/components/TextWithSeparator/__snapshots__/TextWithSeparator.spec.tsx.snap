// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextWithSeparator should render a TextWithSeparator with a custom separator 1`] = `
<div>
  <p
    class="textWithSeparator"
  >
    <span>
      March 08, 2021
    </span>
    <span
      class="textWithSeparator__separator"
    >
      |
    </span>
    <span>
      9 minute read
    </span>
  </p>
</div>
`;

exports[`TextWithSeparator should render the TextWithSeparator component 1`] = `
<div>
  <p
    class="textWithSeparator"
  >
    <span>
      San Mateo, CA
    </span>
    <span
      class="textWithSeparator__separator"
    >
      ▸
    </span>
    <span>
      June 2018 – Present
    </span>
  </p>
</div>
`;
